dbt/adapters/impala/cloudera_tracking.py

```diff
@@ -54,13 +54,22 @@ def track_usage(tracking_payload):
 
     # inner function which actually calls the endpoint
     def _tracking_func(data):
-        SNOWPLOW_ENDPOINT = config("SNOWPLOW_ENDPOINT")
-        SNOWPLOW_TIMEOUT = int(config("SNOWPLOW_TIMEOUT"))  # 10 seconds
+        global usage_tracking
+
+        try:
+            SNOWPLOW_ENDPOINT = config("SNOWPLOW_ENDPOINT")
+            SNOWPLOW_TIMEOUT = int(config("SNOWPLOW_TIMEOUT"))  # 10 seconds
+            SNOWPLOW_API_KEY = config("SNOWPLOW_API_KEY")
+            SNOWPLOW_ENV = config("SNOWPLOW_ENV")
+        except Exception as err:
+            logger.debug(f"Error reading tracking config. {err}")
+            logger.debug("Disabling usage tracking due to error.")
+            usage_tracking = False
```

Comment:
Shouldn't we exit from here if usage_tracking is false. If we have exception in reading config, we should not proceed with the request.

Reply:
No. We should not exit per se. we should return gracefully to dbt main and continue with rest of dbt run instead of failing. I noticed that now. My new pr will address that.